vm/hash.go

```diff
@@ -426,6 +426,45 @@ var builtinHashInstanceMethods = []*BuiltinMethodObject{
 		},
 	},
 	{
+		// Performs a 'shallow' copy of the hash and returns it.
+		// Any arguments are ignored.
+		// The object_id of the returned object is different from the one of the receiver.
+
+		// Caveat: any keys of hash ARE also copied with different object ids for now.
```

Comment:
This shouldn't be the case, @hachi8833 can you open an issue for this and I'll fix it?

Reply:
OK!

Reply:
This won't be solved in the near future. Can you update other comments and I'll merge it?

Reply:
@st0012 how about this?

> 		// Performs a copy of the hash, including the keys and values, and returns it.
>		// Any arguments are ignored.
>		// The object_id of the returned object is different from the one of the receiver.
>		// Caveat: any keys of hash ARE also copied with different object ids for now.
>		// This comes from the fact that the string objects are NOT frozen in current Goby.
>		// TODO: Perhaps the `Hash#dup` will not copy keys when we change the implementation of string objects as frozen.

Reply:
Think the TODO part is unnecessary here. Otherwise looks good.

Reply:
Thnx!